openmdao/devtools/xdsm_viewer/html_writer.py

```diff
@@ -0,0 +1,127 @@
+"""
+HTML file writing to create (semi)standalone XDSMjs output file.
+"""
+
+import json
+import os
+
+from six import iteritems
+
+_DEFAULT_JSON_FILE = "xdsm.json"
+_HTML_TEMPLATE = "index.html"
+
+
+def write_html(outfile, source_data=None, data_file=None):
+    """
+    Writes XDSMjs HTML output file, with style and script files embedded.
+
+    The source data can be the name of a JSON file or a dictionary.
+    If a JSON file name is provided, the file will be referenced in the HTML.
+    If the input is a dictionary, it will be embedded.
+
+    If both data file and source data are given, data file is
+
+    Parameters
+    ----------
+    outfile : str
+        Output HTML file
+    source_data : str or dict or None
+        XDSM data in a dictionary or string
+    data_file : str or None
+        Output HTML file
+    """
+
+    # directories
+    main_dir = os.path.dirname(os.path.abspath(__file__))
+    code_dir = os.path.join(main_dir, 'XDSMjs')
+    src_dir = os.path.join(code_dir, "src")
+    build_dir = os.path.join(code_dir, "build")
+    vis_dir = os.path.join(main_dir, "visualization")
+    style_dir = code_dir  # CSS
+
+    # grab the libraries
+    scripts = ''
+    script_names = {'animation', 'controls', 'graph', 'labelizer', 'selectable', 'xdsm',
+                    'xdsm-factory'}
+
+    with open(os.path.join(build_dir, "xdsm.bundle.js"), "r") as f:
+        code = f.read()
+        xdsm_bundle = _write_script(code, {'type': 'text/javascript'})
+    # grab the scripts
```

Comment:
You don't need to grab the js files from XDSMjs src directory, all you need is the xdsm.bundle.js file which is actually the output of the XDSMjs build process which uses those js source files as inputs.

Reply:
Thanks for the explanation, corrected.